specification/network/resource-manager/Microsoft.Network/2017-06-01/expressRouteConnector.json

```diff
@@ -0,0 +1,372 @@
+					"description": "Request successful. The operation returns the resulting Express Route Connector Resource.",
+					"schema": {
+					  "$ref": "#/definitions/expressRouteConnector"
+					}
+				  },
+				  "201": {
+					"description": "Request successful. The operation returns the resulting Express Route Connector Resource.",
+					"schema": {
+					  "$ref": "#/definitions/expressRouteConnector"
+					}
+				  }
+				},
+				"x-ms-long-running-operation": true
+			}
+		},
+	 "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/expressRouteConnectors": {
+		  "get": {
+			"tags": [
+			  "ExpressRouteConnectors"
+			],
+			"operationId": "ExpressRouteCircuitConnectors_ListByResourceGroup",
+			"x-ms-examples": {
+                    "ExpressRouteConnectorListByResourceGroup": { "$ref": "./examples/ExpressRouteConnectorListByResourceGroup.json" }
+            },
+			"description": "Gets all express route Connector resources in a given resource group and subscription.",
+			"parameters": [
+			  {
+				"$ref": "#/parameters/ResourceGroupNameParameter"
+			  },
+			  {
+				"$ref": "#/parameters/ApiVersionParameter"
+			  },
+			  {
+				"$ref": "#/parameters/SubscriptionIdParameter"
+			  }
+			],
+			"responses": {
+			  "200": {
+				"description": "Request successful. The operation returns a list of Express Route Connector resources.",
+				"schema": {
+				  "$ref": "#/definitions/expressRouteConnectorListResult"
+				}
+			  }
+			},
+			"x-ms-pageable": {
+				"nextLinkName": "nextLink"
+			}
+		  }
+		},	
+	"/subscriptions/{subscriptionId}/providers/Microsoft.Network/expressRouteConnectors": {
+		  "get": {
+			"tags": [
+			  "ExpressRouteConnectors"
+			],
+			"operationId": "ExpressRouteCircuitConnectors_ListBySubscription",
+			"x-ms-examples": {
+                    "ExpressRouteConnectorListBySubscription": { "$ref": "./examples/ExpressRouteConnectorListBySubscription.json" }
+            },
+			"description": "Gets all express route Connector resources in a given subscrption.",
+			"parameters": [
+			  {
+				"$ref": "#/parameters/ApiVersionParameter"
+			  },
+			  {
+				"$ref": "#/parameters/SubscriptionIdParameter"
+			  }
+			],
+			"responses": {
+			  "200": {
+				"description": "Request successful. The operation returns a list of Express Route Connector resources.",
+				"schema": {
+				  "$ref": "#/definitions/expressRouteConnectorListResult"
+				}
+			  }
+			},
+			"x-ms-pageable": {
+				"nextLinkName": "nextLink"
+			}
+		  }
+		}	
+	},	 
+	"definitions": {	
+		"expressRouteConnectorPropertiesFormat": {
+			"required": [
+				"circuits",
+				"addressPrefix"
+			],
+			"properties": {
+				"circuits": {
+					"type": "array",
+					"items": {
+						"$ref": "./network.json#/definitions/SubResource",
+						"description": "The reference to the express route circuit resource."
+					},
+					"description": "List of Express Route Circuits."
+				},
+				"authorizations": {
```

Comment:
The customer specifies authorization Key as part of PUT API Call and we just intend to return whatever customer specified in ExpressRouteConnector object. Authorization Key is also returned as part of VirtualNetworkGatewayConnectionObject : Reference : virtualNetworkGateway.json